src/plugin/adapter-registry/c-tracker.ts

```diff
@@ -42,13 +42,14 @@ export class CTracker extends AdapterVariableTracker {
             return undefined;
         }
         try {
-            const variableAddress = await this.getAddressOfVariable(variable.name, session);
+            const [variableAddress, variableSize] = await Promise.all([
+                variable.memoryReference && hexAddress.test(variable.memoryReference) ? variable.memoryReference : this.getAddressOfVariable(variable.name, session),
```

Comment:
Should we consider different integer representations? Not sure if all debug adapters would always return a hex value for an address (although IMO the sensible thing to do).

Reply:
That is a good point! But I'd rather defer this discussion into a separate issue. The previous behavior also used the `hexAddress`. I did add a commit that unifies that check but I still think it does not necessarily have to be part of this PR, do you agree?

Reply:
Fine with me. TBF, I'd expect addresses to be returned as hex values anyway.

Reply:
Since this isn't standardized, I'd probably prefer a check for any (easily parsed) number format so that `16` is valid in addition to `0x10`. That would follow the convention for things like the [`ReadMemoryResponse`](https://microsoft.github.io/debug-adapter-protocol/specification#Requests_ReadMemory) and [`DisassembledInstruction`](https://microsoft.github.io/debug-adapter-protocol/specification#Types_DisassembledInstruction) where at least hex and decimal are both considered acceptable formats.

Reply:
Fantastic, thank you! I added https://github.com/eclipse-cdt-cloud/vscode-memory-inspector/issues/107 to track this. It shouldn't be too much work but it needs to be discussed first (all formats vs "only" hexadecimal/decimal because they are mentioned in the DAP).